File: python3/1752.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        df = False
        for i in range(1, n):
            if nums[i - 1] > nums[i]:
                if df:
                    return False
                df = True
        return nums[0] >= nums[n - 1] or (not (df))

File: java/1752.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        boolean df = false;
        for (int i = 1; i < n; i++)
            if (nums[i - 1] > nums[i]) {
                if (df)
                    return false;
                df = true;
            }
        return (nums[0] >= nums[n - 1] || !df);
    }
}
